src/Actions/ReplaceFileContentAction.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Regnerisch\LaravelBeyond\Actions;
+
+use Illuminate\Filesystem\Filesystem;
+use Symfony\Component\Filesystem\Exception\FileNotFoundException;
+use Symfony\Component\Filesystem\Exception\IOException;
+
+class ReplaceFileContentAction
+{
+    public function execute(string $path, array $replacements): void
+    {
+        $fs = new Filesystem();
+
+        if (!$fs->exists($path)) {
+            throw new FileNotFoundException("$path not found");
+        }
+
+        if (!$fs->isReadable($path)) {
+            throw new IOException("$path is not readable.");
+        }
+
+        if (!$fs->isWritable($path)) {
+            throw new IOException("$path is not writable.");
+        }
+
+        $content = $fs->get($path);
+
+        foreach ($replacements as $search => $replace) {
```

Comment:
Why not `str_replace(array_keys($replacements), $replacements, $content)`? As it would obmit the foreach

Reply:
resolved, using `RefactorFileAction` from now on